src/cli/expand-patterns.js

```diff
@@ -163,7 +163,11 @@ function containsIgnoredPathSegment(absolutePath, cwd, ignoredDirectories) {
   return path
     .relative(cwd, absolutePath)
     .split(path.sep)
-    .some((dir) => ignoredDirectories[dir]);
+    .some(
```

Comment:
Can you pass a `Set` to this function instead of `object`?

Reply:
I mean turn `silentlyIgnoredDirs` to `Set`

Reply:
Done in https://github.com/prettier/prettier/pull/10256/commits/8fbf36f72c2d1eb98f57039de54ffe1f83acfad3